### PR TITLE
Classic Taskbar Context Menu 1.0.2

### PR DESCRIPTION
**Pull request by ItsProfessional**
Improve code

### CODE REVIEW

File: mods/classic-taskbar-context-menu.wh.cpp [hunk 1]
```diff
@@ -430,6 +431,32 @@ BOOL CALLBACK EnumWindowsCallBack(HWND hwnd, LPARAM lParam) {
...
+        SetWindowSubclassFromAnyThread(hWnd_TaskBar_SecondScreen, SecondTaskbarSubclassProc, 0, 0);
+    }
+
+
+    // Network & Volumn
+    HWND hPNIHiddenWnd = FindWindowA("PNIHiddenWnd", NULL);
+    EnumWindows(EnumWindowsCallBack, GetWindowThreadProcessId(hPNIHiddenWnd, NULL));
+
+    // Notification
+    HWND hNotificationWindow = FindWindowA("NotifyIconOverflowWindow", NULL); // EnumChildWindows
```
Comment (learn-more) [inline mods/classic-taskbar-context-menu.wh.cpp]: You don't use this line

Comment (ItsProfessional) [reply mods/classic-taskbar-context-menu.wh.cpp]: I've removed it

**Comment by m417z** [comment]
You didn't change the version in the metadata.

@learn-more we need to improve the gate to check these things. Are you willing to work on it? If not, I'll try to implement it myself.

**Comment by ItsProfessional** [comment]
I've updated the version number. I'm currently on a phone, so I cannot squash it at the moment. However, I'm pretty sure you should be able to push to my fork. So, I'll leave it to you to squash and force push it.

**Comment by learn-more** [comment]
> You didn't change the version in the metadata.
> 
> @learn-more we need to improve the gate to check these things. Are you willing to work on it? If not, I'll try to implement it myself.

What do you want to check exactly?

**Comment by m417z** [comment]
@learn-more I want to check that the PR adheres to the submission instructions in the readme, and add additional checks too. That's what I can think of right now:

* General
  * First mod line should be `// ==WindhawkMod==`
  * `@id` should match the filename
  * `@id`, `@github` ,`@version`, `@author` should be specified
  * `@id` should be in `([0-9a-z]+-)*[0-9a-z]+` format and have a sane length (<=40 maybe)
  * `@version` should be in `([0-9]+\.)*[0-9]+` format
  * `@compilerOptions` should only specify `-l` and `-D` flags, might add more flags in the future
* Submitting a New Mod (pull request must consist of a single file, `mods/<mod-id>.wh.cpp`)
  * `@author`, `@twitter`, `@homepage` should match previously submitted mods of `@github`, if any
  * `@author`, `@twitter`, `@homepage` should not collide with another submitter's `@author`/`@twitter`/`@homepage`
  * If `@twitter` is specified for the first time, show a warning so I'll know to validate it
* Submitting a Mod Update (pull request must consist of changes to a single file, `mods/<mod-id>.wh.cpp`)
  * `@id`, `@author`, `@github`, `@twitter`, `@homepage` should not change
  * `@version` should be changed to a higher version

The checks aren't necessarily required for the mod to be merged, but I want to be alerted when they don't pass. If they do, I want to be fairly confident I can just merge without looking at it manually too much.

The checks can help avoid mistakes, but the mindset should be that somebody with malicious intentions might try to sneak harmful code, and while it's not feasible to assure 100% of the code of each submitted mod is not malicious, I want to make it as difficult as possible to sneak something in. For example, I want to avoid this kind of a submission:

```cpp
// ==WindhwakMod==
// @id           new-mod
// @metadata1 ...
// @metadata2 ...
// ==WindhwakMod==

// (mod code here)

// ==WindhawkMod==
// @id           new-mod
// @metadata1 ...
// @metadata2 ...
// @compilerOptions -some -malicious -stuff
// ==WindhawkMod==

// (more mod code here)
```

(did you spot the deliberate typo?)

**Comment by learn-more** [comment]
@m417z I think that means basically rewriting the entire validation script (or at least major parts of it).
I have created an issue for this, and I do intend to work on it, but cannot give you a timeframe.

**Comment by m417z** [comment]
Great, thanks.

**Pull request merged**